core/modules/xrdsvc/plugin.cc

```diff
@@ -20,30 +20,57 @@
  * the GNU General Public License along with this program.  If not,
  * see <http://www.lsstcorp.org/LegalNotices/>.
  */
-/// Implement XrdSsiGetServerService() to provide Qserv's SsiService
+/// Implement XrdSsiProviderServer to provide Qserv's SsiService
 /// implementation. Link this file when building a plugin to be used as
-/// ssi.svclib .
-
+/// ssi.svclib.
 
 // System headers
-#include <iostream>
+#include <memory>
 
 // Third-party headers
-#include "XrdSsi/XrdSsiLogger.hh"
-#include "XrdSsi/XrdSsiService.hh"
+#include "XrdSsi/XrdSsiProvider.hh"
 
 // Qserv headers
-#include "xrdsvc/SsiService.h"
+#include "SsiService.h"
 
-class XrdSsiLogger;
-class XrdSsiCluster;
+namespace lsst {
+namespace qserv {
+namespace xrdsvc {
 
-extern "C" {
-XrdSsiService *XrdSsiGetServerService(XrdSsiLogger  *logP,
-                                      XrdSsiCluster *clsP,
-                                      const char    *cfgFn,
-                                      const char    *parms)
+class SsiProviderServer : public XrdSsiProvider
```

Comment:
I'd stay consistent with the rest of qserv: put the bracket on previous line (in many places in this file)